lib/convertor.js

```diff
@@ -98,61 +98,96 @@ class MongoToKnex {
 
         // CASE: fallback, `status=draft` -> `posts.status`=draft
         return {
-            column: this.tableName + '.' + column,
+            column: `${this.tableName}.${column}`,
             value: value,
             isRelation: false
         };
     }
 
     /**
-     * Build queries for relations.
+     * We group the relations by a unique key.
+     * Each grouping will create a sub query.
+     *
+     * @TODO: choose a different key name (!)
+     *
+     * Returns a group structure of following format:
+     *  {
+     *      "groupKey": {
+     *          innerWhereStatements: [],
+     *          joinFilterStatements: []
+     *      }
+     *  }
      */
-    buildRelationQuery(qb, relations) {
-        debug(`(buildRelationQuery)`);
-
-        if (debugExtended.enabled) {
-            debugExtended(`(buildRelationQuery) ${JSON.stringify(relations)}`);
-        }
-
-        const groupedRelations = {};
-
-        /**
-         * We group the relations by a unique key.
-         * Each grouping will create a sub query.
-         *
-         * @TODO: choose a different key name (!)
-         */
-        _.each(relations, (relation) => {
-            if (!groupedRelations[relation.table]) {
-                groupedRelations[relation.table] = [];
-            }
-
-            if (groupedRelations[relation.table].length) {
-                const columnExists = _.find(groupedRelations[relation.table], (statement) => {
-                    /**
-                     * CASE:
-                     * - we should not use the same sub query if the column name is the same (two sub queries)
-                     * - e.g. $and conjunction requires us to use 2 sub queries, because we have to look at each individual tag
-                     */
-                    if (statement.column === relation.column) {
+    groupRelationStatements(statements, mode) {
```

Comment:
This method started as a logic extraction from `buildRelationQuery` and grew with more grouping related logic neede for better $and/$or support